pyiceberg/table/__init__.py

```diff
@@ -2492,16 +2497,28 @@ def _add_and_move_fields(
 class WriteTask:
     write_uuid: uuid.UUID
     task_id: int
+    schema: Schema
     record_batches: List[pa.RecordBatch]
     sort_order_id: Optional[int] = None
+    partition_key: Optional[PartitionKey] = None
 
-    # Later to be extended with partition information
+    def generate_data_file_partition_path(self) -> str:
```

Comment:
Nit: This function looks redundant. The check is being done in `generate_data_file_path()` as well. I would merge those two.